architecture/adr-list/adr-008-estimating-transaction-fees.md

```diff
@@ -0,0 +1,253 @@
+
+If `gas-prices` specified in a transaction is lower than the `min-gas-prices` of _all_ validators, no validators consider that transaction in their mempool. The transaction eventually times out and is not processed.
+
+If `gas_prices` specified in a transaction is lower than `min-gas-prices` of _some_ validators, the transaction has a chance at being committed within the transaction timeout window. Whether this actually happens depends on the sum of voting powers of validators that are ready to commit it.
+
+There is currently no way to request `min-gas-prices`.
+
+#### **`fee`**
+
+`fee` is the amount of tokens a validator takes for a transaction processing and is calculated as `fee` = `gas` \* `gas-prices`
+
+`gas` above must be at least equal to or more than `gas-wanted`. 
+
+### Options considered for estimating transaction fees
+
+#### Gas estimation
+
+**Option 1: Client-side calculation**
+
+1. A gas estimation request could be built into the client libraries \(such as [VDR Tools](https://gitlab.com/evernym/verity/vdr-tools)\) used in applications using cheqd network
+2. The client library would use a precalculated adjustment coefficient to increase the chances of the transaction being committed.
+
+**Option 2: Use estimated values**
+
+1. Estimated gas for most common transactions could be pre-calculated based on best-guess.
+2. This method can be used as a workaround in the absence of a formal gas estimation request in Cosmos.
+3. This approach can be used because gas needed to process simple transactions will remain constant over time.
+
+#### Fee price estimations
+
+**Option 1: Use fixed values**
+
+1. Set a fixed recommended gas price on the cheqd network
+2. Embed the fixed value into applications
+3. This scenario is more likely to fail over time because validator operators can change gas prices.
+
+**Option 2: Dynamically calculate values**
+
+1. Gas prices can be dynamically determined based on recent transactions.
+2. Possible ways of estimation:
+   1. Moving average
+   2. Weighed moving average
+   3. Exponential moving average
+3. Possible ways of implementation:
+   1. Client side
+   2. Provided as a service
+   3. On chain (the most trusted and robust way)
+3. To prevent constant growth of the mean value it's proposed to use multiplier `Y < 1` to get star fee price.
+4. If the dynamically calculated fees are insufficient to put a transaction though, use exponential growth to calculate higher values and retry.
+
+### Proposed transaction sending flow in VDR Tools
+
+1. Build and sign a transaction
+2. Send gas estimation request
+   1. Multiply response to the precalculated coefficient `X > 1` to increase probability of the transaction being committed
+3. Set initial gas price
+   1. Request moving average from the network
+   2. Multiply response by `Y < 1` to prevent constant growth of the mean value
```

Comment:
Yes, this is a good idea. Although how do you want to set this multiplier/factor...random? Or other metric?

Reply:
At this point the proposal is to take either minimum or let's say 5th percentile and multiply it by 0.9. Numbers are taken from the top of my head and are subjects of research or experiment.